dbt_subprojects/dex/tests/generic/test_reasonable_usd_amount.sql

```diff
@@ -0,0 +1,62 @@
+{% test test_reasonable_usd_amount(model, column_name, max_value=1000000000) %}
+
+{# 
+    Test to ensure USD amounts are within a reasonable range
+    Args:
+        model: The model to test
+        column_name: The column containing USD amounts
+        max_value: Maximum allowed USD amount (default: $1B)
+#}
+
+WITH base_trades AS (
+    SELECT
+        blockchain,
+        project,
+        version,
+        block_month,
+        block_date,
+        block_time,
+        block_number,
+        token_bought_amount_raw,
+        token_sold_amount_raw,
+        token_bought_address,
+        token_sold_address,
+        taker,
+        maker,
+        project_contract_address,
+        tx_hash,
+        evt_index,
+        tx_from,
+        tx_to
+    FROM {{ model }}
+    WHERE block_time >= NOW() - INTERVAL '1' day  -- Only check recent trades
```

Comment:
The full test suite for a `1 day`'s data cycle takes approximately 25 minutes to complete, with individual test durations ranging from a few seconds to over 20 minutes. 

The average test runtime is about 5.4 minutes, while some chain-specific tests, particularly for major networks like Ethereum, Avalanche, and BNB Chain, require significantly longer execution times.

Here are some of the most time-consuming tests :

Ethereum: 23 minutes 56 seconds
Avalanche C-Chain: 23 minutes 55 seconds
Celo: 23 minutes 54 seconds
BNB Chain (Binance Smart Chain): 23 minutes 52 seconds
Arbitrum: 22 minutes 49 seconds

https://github.com/duneanalytics/spellbook/actions/runs/12789818470/job/35654151891?pr=7463#step:13:65